spec/amber/controller/respond_with_spec.cr

```diff
@@ -7,6 +7,10 @@ module Amber::Controller
       request.headers["Accept"] = ""
       context = create_context(request)
 
+      Spec.before_each do
+        context.response.status_code = 200
```

Comment:
I don't think you want to do this here... `Spec.before_each` is global and applies to all `it` blocks in the suite, it's not scoped to a describe / context block.

Reply:
@damianham Please review this @robacarp comment, we need this because https://github.com/amberframework/recipes/pull/4 depends on it :sweat_smile:

Reply:
Agree it should not be global but it seems there is no option to run before/after hooks that apply only to the current spec context.  The test suite passes with this in place and fails if I remove it. I did spend some time trying to remove this hook but I couldn't get the test suite to pass so I think it is a case of accept this or reject the PR.  I don't see any real downsides to explicitly setting the status_code to the default status.

Reply:
The downside is that you don't know what tests will be written in the future that this will cause a weird breakage and possibly sink a day of debugging. I'd much rather have this one line duplicated across the tests that need it than have the global leakage.